Orange/classification/_tree_scorers.pyx

```diff
@@ -0,0 +1,293 @@
+#cython: embedsignature=True
+
+import numpy
+cimport numpy as np
+import cython
+
+from libc.stdlib cimport calloc, free
+
+cdef extern from "math.h":
+    double log(double x)
+
+cdef extern from "numpy/npy_math.h":
+    bint npy_isnan(double x)
+
+@cython.boundscheck(False)
+@cython.wraparound(False)
+def find_threshold_entropy(np.ndarray[np.float64_t, ndim=1] x,
+                           np.ndarray[np.float64_t, ndim=1] y,
+                           np.ndarray[np.int64_t, ndim=1] idx,
+                           int n_classes, int min_leaf):
```

Comment:
```py
def find_threshold_entropy(double[:] x,
                           double[:] y,
                           int[:] idx,
                           int n_classes,
                           int min_leaves):
```
I find this syntax somewhat more readable. It is compatible and potentially faster.